### PR TITLE
[DEM-1030] Add __repr__ to configuration helper

### PR DESCRIPTION
**Pull request by qSaevar**

### CODE REVIEW

File: src/tests/unittests/configuration_helper/test_configuration_helper.py [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
 from unittest.mock import MagicMock, Mock, patch
 
-from qilib.configuration_helper import InstrumentConfiguration, InstrumentConfigurationSet
+from qilib.configuration_helper import InstrumentConfiguration, InstrumentConfigurationSet, InstrumentAdapterFactory
 from qilib.configuration_helper.configuration_helper import ConfigurationHelper
 from qilib.utils import PythonJsonStructure
 from qilib.utils.storage import StorageMemory
+from tests.test_data.dummy_instrument_adapter import DummyInstrumentAdapter
+
+InstrumentAdapterFactory.add_instrument_adapter_class(DummyInstrumentAdapter)
```
Comment (CumulonimbusCalvus) [inline src/tests/unittests/configuration_helper/test_configuration_helper.py]: I do not understand why this change has been made. If I remove the changes it all still works. Do you understand why?

Comment (QFer) [reply src/tests/unittests/configuration_helper/test_configuration_helper.py]: You are totally right. I will delete it. I don't know why it was added, maybe for a new test that was deleted afterwards.
